### PR TITLE
Support fixed size byte arrays in database natively

### PR DESCRIPTION
**Pull request by vkgnosis**
without going through Vec<u8>

This is more efficient, more type safe, easier to use.

### Test Plan

new and old unit tests

### CODE REVIEW

File: crates/database/src/lib.rs [hunk 1]
```diff
@@ -20,6 +23,11 @@ pub async fn clear_DANGER(ex: impl PgExecutor<'_> + Copy) -> sqlx::Result<()> {
     Ok(())
 }
 
+pub type Address = ByteArray<20>;
+pub type AppId = ByteArray<32>;
+pub type TransactionHash = ByteArray<32>;
+pub type OrderUid = ByteArray<56>;
```
Comment (vkgnosis) [inline crates/database/src/lib.rs:29]: Not sure if plain aliases or newtypes are better here. Both give type safety and prevent copy paste errors but newtypes gives a little more typesafety while being a little harder to use. For now went with newtypes.

File: crates/database/src/byte_array.rs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    }
+}
+
+impl<const N: usize> Decode<'_, Postgres> for ByteArray<N> {
+    fn decode(value: PgValueRef<'_>) -> Result<Self, BoxDynError> {
+        let mut bytes = [0u8; N];
+        match value.format() {
+            // prepared query
+            PgValueFormat::Binary => {
+                bytes = value.as_bytes()?.try_into().unwrap();
```
Comment (nlordell) [inline crates/database/src/byte_array.rs:29]: Is `unwrap` intentional here? Shouldn't this be an error if you have a binary value with the wrong number of bytes?

**Review by nlordell — COMMENTED** [review]
This is so cool!

Just one concern around the `unwrap` - it is unclear to me why its OK to unwrap in that situation. (Maybe I'm missing context around what a "prepared query" actually is).

File: crates/database/src/lib.rs (hunk 1 above)
Comment (nlordell) [reply crates/database/src/lib.rs:29]: Personally, I prefer type aliases here (unless you intend on adding methods to these).

My reasoning is that this is a representation of the Postgres storage that it is using.

File: crates/database/src/byte_array.rs (hunk 2 above)
Comment (vkgnosis) [reply crates/database/src/byte_array.rs]: That was leftover from testing. Thanks for catching. Adjusted test to make sure it errors in this case.

**Review by nlordell — APPROVED** [review]

**Review by MartinquaXD — APPROVED** [review]
Nice change. 👍

**Pull request merged**